src/access/acp.ts

```diff
@@ -20,24 +20,48 @@
  */
 
 import { WithAccessibleAcr } from "../acp/acp";
-import { getPolicyUrlAll } from "../acp/control";
+import { getAcrPolicyUrlAll, getPolicyUrlAll } from "../acp/control";
 import { internal_getAcr } from "../acp/control.internal";
+import { getPolicyAll } from "../acp/policy";
 import { getRuleAll } from "../acp/rule";
+import { acp } from "../constants";
 import { WithResourceInfo } from "../interfaces";
 import { getSourceIri } from "../resource/resource";
-import { asUrl } from "../thing/thing";
+import { getSolidDataset } from "../resource/solidDataset";
+import { getUrlAll } from "../thing/get";
+import { asUrl, getThing } from "../thing/thing";
 
 export function internal_hasInaccessiblePolicies(
   resource: WithAccessibleAcr & WithResourceInfo
 ): boolean {
   const sourceIri = getSourceIri(resource);
-  const policyUrls = getPolicyUrlAll(resource);
-  const ruleUrls = getRuleAll(internal_getAcr(resource)).map((rule) =>
-    asUrl(rule)
-  );
-  // If either a policy or a rule are not defined in the ACR, return false
+
+  // Collect all policies that apply to the resource or its ACR (aka active)
+  const activePolicyUrls = getPolicyUrlAll(resource);
+  getAcrPolicyUrlAll(resource).forEach((policyUrl) => {
+    activePolicyUrls.push(policyUrl);
+  });
```

Comment:
More a matter of style, but this saves some manual looping:

```suggestion
  const activePolicyUrls = getPolicyUrlAll(resource).concat(getAcrPolicyUrlAll(resource));
```

Reply:
Yeah it looks better, I remember thinking "How nice would it be to have an `addAll` method, or if `push` accepted a whole array", and left it there while I used concat three lines lower ^^.

Resolved in 82181eda8eee05b632e1f18511d407532666e438